services/bundle_analysis.py

```diff
@@ -25,3 +31,108 @@ def load_report(
         repo_key=ArchiveService.get_archive_hash(commit.repository),
     )
     return loader.load(commit_report.external_id)
+
+
+def load_time_conversion(size):
+    """
+    Converts total size in bytes to approximate time (in seconds) to download using a 3G internet (3 Mbps)
+    """
+    return round((8 * size) / (1024 * 1024 * 3), 1)
+
+
+@dataclass
+class BundleAnalysisComparison(object):
+    def __init__(
+        self,
+        loader: BundleAnalysisReportLoader,
+        base_report_key: str,
+        head_report_key: str,
+    ):
+        self.comparison = SharedBundleAnalysisComparison(
+            loader,
+            base_report_key,
+            head_report_key,
+        )
+        self.bundle_comparisons = []
+        self.total_size_delta = 0
+        self.total_size = sum(
+            [
+                report.total_size()
+                for report in self.comparison.head_report.bundle_reports()
+            ]
+        )
+
+        for bundle_change in self.comparison.bundle_changes():
+            self.total_size_delta += bundle_change.size_delta
+
+            if (
+                self.comparison.head_report
+                and self.comparison.head_report.bundle_report(bundle_change.bundle_name)
+            ):
+                head_bundle_report_size = self.comparison.head_report.bundle_report(
```

Comment:
Consider declaring `self.comparison.head_report.bundle_report(..)` as a variable so we're not calling it twice.

Reply:
👍